### PR TITLE
feat: use hashbrown::hashmap everywhere (modestly faster decompress)

### LINKED ISSUE #1050: Create alias type for HashMap and HashSet with our favourite hasher
Right now we use a mixture of std, ahash and foldhash. We should do what Polars did, i.e. define their own alias for HashSet and HashMap with the hasher we like the most and ban usages of std::collections::HashMap/HashSet

### PR DESCRIPTION
**Pull request by danking**
Resolves #1050 .

I saw a modest (~5%) reduction in decompress time on a string-heavy dataset in which SparseArray take occupies ~10% of total time.

### CODE REVIEW

File: clippy.toml [hunk 1]
```diff
@@ -1,3 +1,6 @@
 allow-expect-in-tests = true
 allow-unwrap-in-tests = true
 single-char-binding-names-threshold = 2
+disallowed-types = [
+    "std::collections::HashMap",
```
Comment (AdamGS) [inline clippy.toml:5]: lets add `HashSet` as well

**Comment by robert3005** [comment]
You have stumbled upon something I had complained about before https://github.com/spiraldb/vortex/issues/1050

File: vortex-serde/src/layouts/read/cache.rs [hunk 2]
```diff
@@ -1,9 +1,9 @@
 use std::sync::{Arc, RwLock};
 
-use ahash::HashMap;
```
Comment (robert3005) [inline vortex-serde/src/layouts/read/cache.rs:3]: You can probably remove ahash explicit dependency here

File: vortex-array/src/aliases/mod.rs [hunk 3]
```diff
@@ -0,0 +1 @@
+pub mod hash_map;
```
Comment (AdamGS) [inline vortex-array/src/aliases/mod.rs:1]: `aliases` feels a bit weird, but I don't have a better idea and it does make sense to have a module to contain the type aliases.

**Review by AdamGS — APPROVED** [review]

File: vortex-serde/src/layouts/read/cache.rs (hunk 2 above)
Comment (danking) [reply vortex-serde/src/layouts/read/cache.rs:3]: Done.

**Comment by danking** [comment]
no pushdown becomes weirdly slower? 1.38x and 1.28x

**Review by robert3005 — APPROVED** [review]

**Pull request merged**